api/agent/func_logger.go

```diff
@@ -148,8 +120,9 @@ func (li *lineWriter) Write(ogb []byte) (int, error) {
 		// write in this line and advance buffer past it
 		l := b[:i+1]
 		ns, err := li.w.Write(l)
+		n += ns
 		if err != nil {
-			return ns, err
+			return n, err
 		}
 		li.b.Next(len(l))
```

Comment:
this also looks odd. if li.w.Write() return error with partial write, then we do need to execute li.b.Next(ns) regardless of error, no?